javascript/ql/src/semmle/javascript/MembershipTests.qll

```diff
@@ -73,185 +73,183 @@ module MembershipTest {
 }
 
 /**
- * An `InclusionTest` viewed as a `MembershipTest`.
+ * Provides classes that implement `MembershipTest`.
  */
-class OrdinaryInclusionTest extends InclusionTest, MembershipTest::Range {
-  override DataFlow::Node getCandidate() { result = this.getContainedNode() }
-
-  override boolean getPolarity() { result = InclusionTest.super.getPolarity() }
-}
-
-/**
- * A test for whether a candidate is a member of an array.
- */
-class ArrayMembershipTest extends OrdinaryInclusionTest {
-  DataFlow::ArrayCreationNode array;
+module MembershipTestImplementations {
```

Comment:
I'm a little sad to see yet another naming convention emerge here.

This is public API (it's used by a query) so it shouldn't be half-hidden behind a name like "implementations". Looking at the use-case in the prototype pollution query, it doesn't strike me as an implementation detail that there are different kinds of membership tests, so I think we should just merge it with the `MembershipTest` module itself.

Reply:
Good point. I have merged the two modules.